include/seqan3/alphabet/nucleotide/dna5.hpp

```diff
@@ -69,174 +70,72 @@ namespace seqan3
-    {
-        return other_nucl_type{_value};
-    }
-
-    //!\brief Explicit conversion to any other nucleotide alphabet (via char representation).
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type>
-    //!\endcond
-    explicit constexpr operator other_nucl_type() const noexcept
-    {
-        return detail::convert_through_char_representation<other_nucl_type, std::decay_t<decltype(*this)>>[to_rank()];
-    }
-    //!\}
-
-    //!\name Comparison operators
-    //!\{
-    constexpr bool operator==(dna5 const & rhs) const noexcept
-    {
-        return _value == rhs._value;
-    }
-
-    constexpr bool operator!=(dna5 const & rhs) const noexcept
-    {
-        return _value != rhs._value;
-    }
-
-    constexpr bool operator<(dna5 const & rhs) const noexcept
-    {
-        return _value < rhs._value;
-    }
-
-    constexpr bool operator>(dna5 const & rhs) const noexcept
-    {
-        return _value > rhs._value;
-    }
-
-    constexpr bool operator<=(dna5 const & rhs) const noexcept
-    {
-        return _value <= rhs._value;
-    }
-
-    constexpr bool operator>=(dna5 const & rhs) const noexcept
-    {
-        return _value >= rhs._value;
+    constexpr dna5() : base_t{} {}
+    constexpr dna5(dna5 const &) = default;
+    constexpr dna5(dna5 &&) = default;
+    constexpr dna5 & operator=(dna5 const &) = default;
+    constexpr dna5 & operator=(dna5 &&) = default;
+
+    //!\brief Allow implicit construction from dna/rna of the same size.
+    template <std::Same<rna5> t>    // Accept incomplete type
+    constexpr dna5(t const & r)
+    {
+        using seqan3::to_rank;
+        assign_rank(to_rank(r));
     }
     //!\}
 
 protected:
     //!\privatesection
-    //!\copydoc seqan3::dna4::internal_type
-    enum struct internal_type : rank_type
-    {
-        A,
-        C,
-        G,
-        T,
-        N,
-        U = T,
-        UNKNOWN = N
-    };
 
-    //!\copydoc seqan3::dna4::value_to_char
-    static constexpr char_type value_to_char[value_size]
+    //!\copydoc seqan3::dna4::rank_to_char
+    static constexpr char_type rank_to_char[value_size]
     {
         'A',
         'C',
         'G',
-        'T',
-        'N'
+        'N',
+        'T'
     };
 
-    //!\copydoc seqan3::dna4::char_to_value
-    static constexpr std::array<internal_type, 256> char_to_value
+    //!\copydoc seqan3::dna4::char_to_rank
+    static constexpr std::array<rank_type, 256> char_to_rank
     {
         [] () constexpr
         {
-            using in_t = internal_type;
-            std::array<in_t, 256> ret{};
+            std::array<rank_type, 256> ret{};
 
             // initialize with UNKNOWN (std::array::fill unfortunately not constexpr)
             for (auto & c : ret)
-                c = in_t::UNKNOWN;
+                c = 3; // == 'N'
```

Comment:
I would use `base_t::UNKOWN`just in case we do change the order again..

Reply:
The problem is that this is not yet fully defined. I would have to initialise the struct outside to do this. But it's not too big a problem I think, because this function is now the only place where the actual relation of char and rank number is defined so if we change it, we definetely touch this function anyway!